client/src/views/users/UpdatePasswordModal.vue

```diff
@@ -0,0 +1,211 @@
+          {{ getTitle() }}
+        </ion-title>
+      </ion-header>
+
+      <div class="modal-content inner-content">
+        <div
+          v-show="pageStep === ChangePasswordStep.OldPassword"
+          class="step"
+        >
+          <ms-password-input
+            v-model="oldPassword"
+            @change="updateError"
+            :label="$t('Password.currentPassword')"
+            @on-enter-keyup="nextStep()"
+            :password-is-invalid="passwordIsInvalid"
+            :error-message="errorMessage"
+            ref="currentPasswordInput"
+          />
+        </div>
+        <div
+          v-show="pageStep === ChangePasswordStep.NewPassword"
+          class="step"
+        >
+          <ms-choose-password-input
+            :password-label="$t('Password.newPassword')"
+            ref="choosePasswordInput"
+          />
+        </div>
+      </div>
+
+      <ion-footer class="modal-footer">
+        <ion-buttons
+          slot="primary"
+          class="modal-footer-buttons"
+        >
+          <ion-button
+            fill="clear"
+            size="default"
+            @click="cancel"
+          >
+            {{ $t('ContactDetailsPage.cancelButton') }}
+          </ion-button>
+          <ion-button
+            fill="solid"
+            size="default"
+            id="next-button"
+            @click="nextStep"
+            :disabled="!canGoForward"
+          >
+            {{ getNextButtonText() }}
+          </ion-button>
+        </ion-buttons>
+      </ion-footer>
+    </div>
+  </ion-page>
+</template>
+
+<script setup lang="ts">
+import { asyncComputed } from '@/common/asyncComputed';
+import { MsChoosePasswordInput, MsModalResult, MsPasswordInput } from '@/components/core';
+import {
+  AvailableDevice,
+  ClientChangeAuthenticationErrorTag,
+  getCurrentAvailableDevice,
+  changePassword as parsecChangePassword,
+} from '@/parsec';
+import { Information, InformationKey, InformationLevel, InformationManager, PresentationMode } from '@/services/informationManager';
+import { translate } from '@/services/translation';
+import { IonButton, IonButtons, IonFooter, IonHeader, IonIcon, IonPage, IonTitle, modalController } from '@ionic/vue';
+import { close } from 'ionicons/icons';
+import { Ref, inject, onMounted, ref } from 'vue';
+
+enum ChangePasswordStep {
+  OldPassword,
+  NewPassword,
+}
+
+const currentDevice: Ref<AvailableDevice | null> = ref(null);
+const informationManager = inject(InformationKey) as InformationManager;
+const pageStep = ref(ChangePasswordStep.OldPassword);
+const choosePasswordInput = ref();
+const oldPassword = ref('');
+const errorMessage = ref('');
+const passwordIsInvalid = ref(false);
+const currentPasswordInput = ref();
+
+onMounted(async () => {
+  await currentPasswordInput.value.setFocus();
+  const deviceResult = await getCurrentAvailableDevice();
+  currentDevice.value = deviceResult.ok ? deviceResult.value : null;
```

Comment:
It's probably better to display a toast and dismiss the modal. We should always be able to get the current device, but in case of a solar eruption, the user will be blocked without knowing what to do. You can just reuse `'ContactDetailsPage.errors.cannotChangePassword'`.